### PR TITLE
Contact us add changes

### PR DESCRIPTION
**Pull request by littlepony99**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -232,9 +234,20 @@
...
+            <version>${thymeleaf.version}</version>
         </dependency>
 
-    </dependencies>
+        <!-- CAPTCHA -->
+<!--        <dependency>-->
+<!--            <groupId>com.octo.captcha</groupId>-->
+<!--            <artifactId>jcaptcha</artifactId>-->
+<!--            <version>${jcaptcha.version}</version>-->
+<!--        </dependency>-->
```
Comment (AlexKievDeveloper) [inline pom.xml]: Please remove commented code

File: src/main/java/com/vinylteam/vinyl/service/CaptchaService.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.vinylteam.vinyl.service;
+
+public interface CaptchaService {
+    byte[] getCaptcha(String captchaId);
+
+    boolean validateCaptcha(String captchaId, String response);
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/service/CaptchaService.java:7]: Please add empty row after interface/class declaration and before closing. It is our code style

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultCaptchaService.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    public boolean validateCaptcha(String captchaId, String response) {
+        return CaptchaValidator.getInstance().validateResponseForID(captchaId,
+                response);
+    }
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultCaptchaService.java:32]: Please add empty row before closing class

File: src/main/java/com/vinylteam/vinyl/util/CaptchaValidator.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.octo.captcha.service.image.DefaultManageableImageCaptchaService;
+import com.octo.captcha.service.image.ImageCaptchaService;
+
+public class CaptchaValidator {
+    private static ImageCaptchaService instance = new DefaultManageableImageCaptchaService();
+
+    public static ImageCaptchaService getInstance() {
+        return instance;
+    }
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/util/CaptchaValidator.java]: Please add empty row after interface/class declaration and before closing. It is our code style

File: src/main/java/com/vinylteam/vinyl/web/servlets/ImageCaptchaServlet.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import com.vinylteam.vinyl.service.impl.DefaultCaptchaService;
+import jakarta.servlet.ServletOutputStream;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+
+public class ImageCaptchaServlet extends HttpServlet {
+    DefaultCaptchaService captchaService = new DefaultCaptchaService();
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/web/servlets/ImageCaptchaServlet.java]: Please add private access modifier

File: src/main/java/com/vinylteam/vinyl/web/servlets/ImageCaptchaServlet.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        byte[] captchaChallengeAsJpeg = captchaService.getCaptcha(captchaId);
+
+        response.setContentType("image/jpeg");
+        response.setStatus(HttpServletResponse.SC_OK);
+        ServletOutputStream responseOutputStream =
+                response.getOutputStream();
+        responseOutputStream.write(captchaChallengeAsJpeg);
+        responseOutputStream.flush();
+        responseOutputStream.close();
+    }
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/web/servlets/ImageCaptchaServlet.java:26]: Please add empty row after interface/class declaration and before closing. It is our code style

**Review by AlexKievDeveloper — CHANGES_REQUESTED** [review]

File: src/test/java/com/vinylteam/vinyl/dao/jdbc/mapper/UserPostRowMapperTest.java [hunk 7]
```diff
@@ -18,20 +20,22 @@
     @Test
     @DisplayName("Map fields correctly for user post conversion")
     void mapFilledRowTest() throws SQLException {
+        LocalDateTime createdAt = LocalDateTime.of(2021, 5, 19, 21, 0);
```
Comment (TEdwardJ) [inline src/test/java/com/vinylteam/vinyl/dao/jdbc/mapper/UserPostRowMapperTest.java]: is there sense to use  now() method too?
Just in order to execute tests on the more wide variety of input data?

**Review by TEdwardJ — DISMISSED** [review]

File: src/main/java/com/vinylteam/vinyl/util/CaptchaValidator.java [hunk 8]
```diff
@@ -4,9 +4,11 @@
 import com.octo.captcha.service.image.ImageCaptchaService;
 
 public class CaptchaValidator {
+
     private static ImageCaptchaService instance = new DefaultManageableImageCaptchaService();
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/util/CaptchaValidator.java]: Maybe should rename variable to plain: "imageCaptchaService" ?

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultCaptchaService.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    @Override
+    public byte[] getCaptcha(String captchaId) {
+        try (ByteArrayOutputStream jpegOutputStream = new ByteArrayOutputStream()) {
+            BufferedImage challenge =
+                    CaptchaValidator.getInstance().getImageChallengeForID(captchaId);
+
+            ImageIO.write(challenge, "jpeg", jpegOutputStream);
+            return jpegOutputStream.toByteArray();
+        } catch (Exception e) {
+            log.error("Error during captcha generation!");
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultCaptchaService.java]: Lets add exception to logger: log.error("Error during captcha generation!", e);

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultCaptchaService.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+            e.printStackTrace();
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultCaptchaService.java]: Maybe we should throw new RuntimeException for stop request processing?

File: src/main/java/com/vinylteam/vinyl/service/impl/DefaultCaptchaService.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+        }
+        return null;
```
Comment (AlexKievDeveloper) [inline src/main/java/com/vinylteam/vinyl/service/impl/DefaultCaptchaService.java]: Please don`t return null. Use Optional for avoid NullPointerException

**Review by AlexKievDeveloper — APPROVED** [review]

**Review by AlexKievDeveloper — APPROVED** [review]

**Review by TEdwardJ — APPROVED** [review]

**Pull request merged**